controls.py

```diff
@@ -0,0 +1,78 @@
+import keyboard
+import pygame
+import sys
+
+class Controls:
+    def __init__(self, comms, exit_program):
+        self.comms = comms
+        self.exit_program = exit_program
+        
+        self.running = True
+        self.SCREEN_COLOR = (0, 0, 0)
+        self.SCREEN_SIZE = (100, 100)
+        
+        pygame.init()
+        self.screen = pygame.display.set_mode(self.SCREEN_SIZE)
+        self.screen.fill(self.SCREEN_COLOR)
+    
+    def on_trigger(self, key):
+        self.comms.read_send(key)
+        print("Sent")
+
+    def run(self):
```

Comment:
This function is a bit too big. Take the code under the main `if`/`else` branches and put them in functions, then use those functions in this function.

Reply:
Lines 29-41 -> function
Lines 42-55 -> function
Lines 58-69 -> function